frankenphp.go

```diff
@@ -539,44 +540,81 @@ func go_ub_write(rh C.uintptr_t, cBuf *C.char, length C.int) (C.size_t, C.bool)
 	return C.size_t(i), C.bool(clientHasClosed(r))
 }
 
+func createHeaderKeyCache() otter.Cache[string, string] {
+	c, err := otter.MustBuilder[string, string](256).Build()
+	if err != nil {
+		panic(err)
+	}
+
+	return c
+}
+
+var headerKeyCache = createHeaderKeyCache()
+
+// There are aroung 60 common request headers according to https://en.wikipedia.org/wiki/List_of_HTTP_header_fields#Request_fields
+// Give some space for custom headers
+
 //export go_register_variables
 func go_register_variables(rh C.uintptr_t, trackVarsArray *C.zval) {
```

Comment:
In this function, it looks like we are doing a ton of unsafe things. Are the perf gains worth managing this much manually?

Reply:
Yes, this prevents copying all headers from the Go memory to the C memory (and then copied again from C memory to C memory because of how PHP works internally). This should also put less pressure on the garbage collector.

According to our benchmark, this does much less allocations. I still need to investigate how this will improve memory usage.

Reply:
The problem may be related to this `free()` anyway because commenting it makes the test suite green. Or it's a race condition of some sort that doesn't always happen.